### PR TITLE
Grafana UI: Add scroll handlers to the Select component

### PR DESCRIPTION
**Pull request by 2Steaks**
### What is this feature?

Adds support for menu scrolling events to the Select component.

https://user-images.githubusercontent.com/4395376/226413792-87f57488-c6cd-4894-9c9a-77928d26b45a.mov

### Why do we need this feature?

The k6 plugin uses paginated data to populate a select dropdown menu with test comparison options, we would like to progressively load in additional options as the user scrolls the menu.

https://user-images.githubusercontent.com/4395376/226414795-7ba83d7d-3f1d-4071-b074-e6649f772446.mov

### CODE REVIEW

File: packages/grafana-ui/src/components/Select/SelectBase.tsx
Comment (tskarhed) [inline packages/grafana-ui/src/components/Select/SelectBase.tsx:218]:
```suggestion
    captureMenuScroll: onMenuScrollToBottom || onMenuScrollToTop,
```
Maybe do this, so we don't have to expose the extra `captureMenuScroll` prop. If you have any reason for not doing this, please do tell! :slightly_smiling_face:

File: packages/grafana-ui/src/components/Select/types.ts
Comment (tskarhed) [inline packages/grafana-ui/src/components/Select/types.ts:73]: I know a lot of props aren't documented for this component, but could you maybe add some docstrings to these props?

File: packages/grafana-ui/src/components/Select/types.ts
Comment (tskarhed) [inline packages/grafana-ui/src/components/Select/types.ts:26]:
```suggestion
```
With the changes above

File: packages/grafana-ui/src/components/Select/SelectBase.tsx
Comment (tskarhed) [inline packages/grafana-ui/src/components/Select/SelectBase.tsx:97]:
```suggestion
```

**Review by tskarhed — CHANGES_REQUESTED** [review]
The Select component has historically been a bit messy, so I want someone else to also give some feedback on this.

**Review by tskarhed — APPROVED** [review]
Sorry for not getting to this sooner! LGTM! :+1:

**Comment by tskarhed** [comment]
@2Steaks Do you want to go ahead and merge this?

**Comment by 2Steaks** [comment]
> @2Steaks Do you want to go ahead and merge this?

@tskarhed it seems I am not an authorised user so I can't merge it.

**Pull request merged**
